cluster-backup-tool.md

```diff
@@ -0,0 +1,144 @@
+server contains these resources on it's classpath, failing fast if they are not present. Finally, the cache metadata is a
+map of all cache names and their associated import file that are contained in the backup.
+
+> A key assumption is that in order to import a data backup to Infinispan version `x`, it's necessary for version `x` of
+the CLI to also be used.
+
+### Format
+Basic json file with key/values, allowing for future changes to ordering and for the GlobalConfiguration json to be
+nested.
+```json
+{
+	"version": "10.1.1.Final",
+	"global-configuration": "...",
+	"caches": ["org.infinispan.CONFIG", "example-user-cache"],
+    "counters: ["example-counter"]
+}
+
+```
+
+## Cache Files
+A cache backup consists of a `.json` and `.dat` file, each of which is named after the cache. The `.json` file is a json
+representation of the cache's configuration and the `.dat` file is the batch file executed by the CLI as part of the import.
+Below is an example `.dat` file for a cache with the name ``example-user-cache`:
+
+1. Create the cache using the `.json` file:
+
+```
+create cache --file=example-user-cache.json example-user-cache
+```
+2. Process the `example-user-cache.dat` batch file:
+
+```
+cd caches/example-user-cache
+put hello world
+put hola mundo
+```
+> If the cache is empty during the backup, then no `.dat` file is created.
+
+### Internal Caches
+Volatile internal caches should not be included in a backup. Internal caches can be excluded during the creation of the
+backup archive, with no `.json` or `.dat` files created for the blacklisted caches.
+
+The counter cache should also be omitted as CLI import statements in the `counters.dat` file should be used instead. 
+
+### Storage MediaTypes
+If a cache has a configured key and/or value MediaType, that can be readily represented as a string, e.g. text/plain or json,
+then the `cache put` batch commands should utilise that encoding. Otherwise a binary MediaType, such
+as `application/octet-stream` is used and the key/value represented as Base64.
+
+For example, the following batch command would be used to store an try in a cache with keys set to `text/plain` and value
+`application/json`:
+```
+cache put --encoding=application/json '{"some":"value"}' key
+```
+
+Or to put an entry with unknown key/values
+```
+cache put --key-encoding=application/octet-stream --value-encoding=application/octet-stream a2V5Cg== c29tZSB2YWx1ZQ==   
+```
+
+## Counters File
+The `counters.dat` file that contains the CLI commands required in order to recreate all counters and their values
+at backup time. If no counters exist, this file can be omitted from the archive.
+
+```
+create counter --initial-value=<value> --storage=<PERSISTENT|VOLATILE> --type=<STRONG|WEAK> <name>
+```
+
+## Backup Archive Format
+The backup files should be packaged as a single archive distribution to aid backup/import.
+
+> Is `.zip` sufficient?
```

Comment:
I would say so for our Windows bretheren.

Reply:
Good point :+1: 

I'll look at how Tristan has handled  this for the server report.